waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/DatabaseMappingUtils.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (C) 2016-2019 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.waggledance.mapping.service;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.hadoop.hive.metastore.api.TableMeta;
+import org.slf4j.Logger;
+
+import com.hotels.bdp.waggledance.mapping.model.DatabaseMapping;
+import com.hotels.bdp.waggledance.util.Whitelist;
+
+public class DatabaseMappingUtils {
+
+  public static final String MANUAL_RESOLUTION_TYPE = "MANUAL";
+  public static final String PREFIXED_RESOLUTION_TYPE = "PREFIXED";
+  private static final String INTERRUPTED_MESSAGE = "Interrupted ";
+  private static final String SLOW_METASTORE_MESSAGE = "Metastore was slow to respond so results are omitted";
+  private static final String NULL_POINTER_MESSAGE = "Got nullpointer";
+
+  public static boolean isWhitelisted(String databasePrefix, String database, Map<String, Whitelist> mappedDbByPrefix) {
+    Whitelist whitelist = mappedDbByPrefix.get(databasePrefix);
+    if ((whitelist == null) || whitelist.isEmpty()) {
+      // Accept everything
+      return true;
+    }
+    return whitelist.contains(database);
+  }
+
+  public static List<String> getMappedWhitelistedDatabases(List<String> databases, DatabaseMapping mapping,
+                                                           Map<String, Whitelist> mappedDbByPrefix) {
+    List<String> mappedDatabases = new ArrayList<>();
+    for (String database : databases) {
+      if (DatabaseMappingUtils.isWhitelisted(mapping.getDatabasePrefix(), database, mappedDbByPrefix)) {
+        mappedDatabases.add(mapping.transformOutboundDatabaseName(database));
+      }
+    }
+    return mappedDatabases;
+  }
+
+  public static void shutdownExecutorService(ExecutorService executorService) {
```

Comment:
This class is a bit of "grab bag" of functionality. The methods above are all about managing the whitelist. This one seems to be handling some boiler plate for closing an `ExecutorService` (do we even need this? is it called from more than one place?). Then the methods below seem to handle getting the results of async calls. I'd suggest splitting this into at least two classes - one with the whitelist functionality, the other with the future functionality.